### PR TITLE
Skip profiling for task classes without profiling information

### PR DESCRIPTION
**Pull request by DSMishler**
This PR fixes the issues related to profiling of tasks without profiling information. We decided to allow PINS modules to generate interesting information even for non-profilable tasks. Thus, we had to make sure that PINS module generating profile information would skip tasks marked without profiling (property `profile=off` in PTG as an example).

Added -2 as another value for the profiling array. `profiling_array[0] == -2` signifies that nothing has been initialized yet.

Replacing both #561 and #573.

### CODE REVIEW

**Review by therault — APPROVED** [review]
LGTM

**Review by devreal — CHANGES_REQUESTED** [review]
@DSMishler I can't judge the fix, but: could you please change the commit message to something more descriptive? And if it is two independent changes maybe split them? Also, can we give `-2` a name to avoid magic numbers?

**Review by bosilca — APPROVED** [review]

**Comment by bosilca** [comment]
@DSMishler I merged the PR that fixes the profiling tests. Please rebase this PR to current HEAD and wait for the CI to complete a test before merging.

**Pull request merged**
